### PR TITLE
feat(pie-radio-group): DSW-2632 add keyboard navigation and focusing

### PR DESCRIPTION
**Pull request by jamieomaguire**
## Describe your changes (can list changeset entries if preferable)
- Introduces keyboard navigation inside the Radio Group component. This functionality should adhere to the behaviours outlined in [this W3 guide](https://www.w3.org/WAI/ARIA/apg/patterns/radio/). Image attached below showing the relevant section.
- I have opted to mirror the `Shift + Tab` behaviour displayed in Chrome, whereby focusing the radio group when no option is selected will focus the last item.
- RTL support is included. When in RTL, the left arrow key will go forwards and the right array will go backwards.

Screenshot 2024-12-12 at 15 45 56

## Author Checklist (complete before requesting a review, do not delete any)
- [x] I have performed a self-review of my code.
- [x] I have added thorough tests where applicable (unit / component / visual).
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview.
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have tested these changes in [PIE Aperture](https://github.com/justeattakeaway/pie-aperture/) using the `/test-aperture` command.

## Not-applicable Checklist items
Please move any Author checklist items that do not apply to this pull request here.
- [ ] I have reviewed visual test updates properly before approving.
---

## Testing
[How do I test my changes?](https://github.com/justeattakeaway/pie/wiki/PIE-Aperture)

| Task                   | Link                             |
|------------------------|----------------------------------|
| Aperture PR            | [🔗](https://github.com/justeattakeaway/pie-aperture/pull/232) |
| NextJS 14 deployment   | [🔗](https://pr232-aperture-nextjs-app-v14.pie.design/components/radio-group) |
| Nuxt 3 deployment      | [🔗](https://pr232-aperture-nuxt-app.pie.design/components/radio-group) |
| Vanilla deployment     | [🔗](https://pr232-aperture-vanilla-app.pie.design/components/radio-group.html) |
| Storybook Deployment | [🔗](https://pr2108-storybook.pie.design/?path=/story/radio-group--default)

## Reviewer checklists (complete before approving)
Mark items as `[-] N/A` if not applicable.

### Reviewer 1 - @fernandofranca 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview.
- [x] I have verified that all acceptance criteria for this ticket have been completed.
- [x] I have reviewed the Aperture changes (if added)
- [-] N/A If there are visual test updates, I have reviewed them.

### Reviewer 2
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview.
- [x] I have verified that all acceptance criteria for this ticket have been completed.
- [x] I have reviewed the Aperture changes (if added)
- [-] N/A If there are visual test updates, I have reviewed them.

### CODE REVIEW

File: packages/components/pie-radio-group/src/index.ts [hunk 1]
```diff
@@ -147,17 +160,262 @@ export class PieRadioGroup extends FormControlMixin(RtlMixin(LitElement)) implem
...
+     * Handles the `focusin` event to manage focus within the radio group.
+     *
+     * This method determines the appropriate element to focus when the radio group
+     * gains focus. It considers the last navigation action (whether `Shift+Tab` was used)
+     * and focuses the checked option, the first option, or the last option as needed.
+     *
+     * **Behaviour:**
+     * - If the event target is the radio group itself:
+     *   - If an option is already checked, it focuses the checked option.
+     *   - If `Shift+Tab` was pressed, it focuses the last option.
```
Comment (xander-marjoram) [inline packages/components/pie-radio-group/src/index.ts:212]: Where did this requirement come from, out of interest?

Comment (xander-marjoram) [reply packages/components/pie-radio-group/src/index.ts:212]: Actually I see it's mimicking native behaviour

Comment (xander-marjoram) [reply packages/components/pie-radio-group/src/index.ts:212]: However focusing the checked option doesn't seem to be native behaviour, unless you saw differently?

Comment (jamieomaguire) [reply packages/components/pie-radio-group/src/index.ts:212]: It's part of the recommended pattern from W3C:

https://www.w3.org/WAI/ARIA/apg/patterns/radio/

Screenshot 2024-12-12 at 11 23 40

Comment (jamieomaguire) [reply packages/components/pie-radio-group/src/index.ts:212]: I have been comparing against this [native example in codepen](https://codepen.io/JamieMaguire/pen/VYZZoYz)

File: packages/components/pie-radio-group/src/index.ts [hunk 2]
```diff
@@ -147,12 +160,152 @@ export class PieRadioGroup extends FormControlMixin(RtlMixin(LitElement)) implem
         }
     }
 
+    protected firstUpdated (): void {
+        // Make all radios impossible to tab to
```
Comment (fernandofranca) [inline packages/components/pie-radio-group/src/index.ts:164]: It'd be nice to add to the comment why it is necessary.

**Review by fernandofranca — DISMISSED** [review]

File: packages/components/pie-radio-group/src/index.ts (hunk 2 above)
Comment (jamieomaguire) [reply packages/components/pie-radio-group/src/index.ts:164]: I can update that now - thanks!

**Review by fernandofranca — APPROVED** [review]
Awesome stuff!!!

File: packages/components/pie-radio-group/test/component/pie-radio-group-new.spec.ts [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+            3: 'radio-8',
+            4: 'radio-9',
+            5: 'radio-10',
+        },
+    },
+    button4: 'btn-4',
+};
+
+test.describe('PieRadioGroup - Component tests new', () => {
+    let pageObject;
```
Comment (JoshuaNg2332) [inline packages/components/pie-radio-group/test/component/pie-radio-group-new.spec.ts:33]: Not sure on your thoughts, but I think we could possibly change this from `pageObject` to something like, `pieRadioGroupPage` or `radioGroupPage` as then when you look at the load functions below, I think it is clearer what is trying to be done.

e.g. `await radioGroupPage.load()`

Comment (JoshuaNg2332) [reply packages/components/pie-radio-group/test/component/pie-radio-group-new.spec.ts:33]: We can also totally address this in a follow up PR.

Comment (jamieomaguire) [reply packages/components/pie-radio-group/test/component/pie-radio-group-new.spec.ts:33]: cheers mate! happy to merge in a follow up if that's ok :)

**Review by JoshuaNg2332 — APPROVED** [review]

**Pull request merged**